### DISCUSSION TITLE
Find the saddle point of a scalar field

### DISCUSSION THREAD

**Discussion by qiqigit** [libigl/libigl#2202] [Q&A]
I'm trying to find the saddle points for a scalar field defined on a triangular mesh. I've tried to call the grad function twice to compute the Hessian matrix, which seems to result in some NaN entries in the matrix. 

I'm wondering if I'm doing it the right way. Any suggestion would be appreciated.

**Comment by alecjacobson** [comment]
Theres different ways to define discrete topology. Checkout "discrete Morse smale complex" (we don't have that in libigl but it would be a cool addition)
